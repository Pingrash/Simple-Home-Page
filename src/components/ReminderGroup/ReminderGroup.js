import React from 'react';
import PropTypes from 'prop-types';
import Reminder from '../Reminder/Reminder';

import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  noReminder: {
    textAlign: 'center',
    margin: '2rem'
  }
}));

const ReminderGroup = ({
  reminders,
  addReminder,
  deleteReminder
}) => {
  const classes = useStyles();

  const remindersToRender = reminders.map(reminder => {
    return (
      <Reminder
        title={reminder.title}
        body={reminder.body}
        addReminder={addReminder}
        deleteReminder={deleteReminder}
        key={reminder.title}
        data-test='reminder'
      />
    );
  });

  return (
    <div>
      <Typography component='h3' variant='h4' data-test='title'>
        Your Reminders
      </Typography>
      {remindersToRender.length > 0 ? (
        remindersToRender
      ) : (
        <Typography
          component='span'
          variant='h5'
          className={classes.noReminder}
          data-test='no-reminders-msg'
        >
          No Reminders
        </Typography>
      )}
    </div>
  );
};

ReminderGroup.propTypes = {
  reminders: PropTypes.arrayOf(PropTypes.object).isRequired,
  addReminder: PropTypes.func.isRequired,
  deleteReminder: PropTypes.func.isRequired
};

export default ReminderGroup;
