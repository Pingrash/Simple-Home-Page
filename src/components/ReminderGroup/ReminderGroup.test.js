import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {
  findByDataTest,
  checkProps,
  testFunc
} from '../../../Utilities';
import ReminderGroup from './ReminderGroup';

const initialProps = {
  reminders: [
    { title: 'title1', body: 'body1' },
    { title: 'title2', body: 'body2' },
    { title: 'title3', body: 'body3' }
  ],
  addReminder: testFunc,
  deleteReminder: testFunc
};

const setup = (props = { ...initialProps }) => {
  const component = shallow(<ReminderGroup {...props} />);
  return component;
};

describe('Snapshot', () => {
  it('Should match snapshot', () => {
    const tree = renderer
      .create(<ReminderGroup {...initialProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ReminderGroup renders correctly with reminders', () => {
  let wrapper;
  beforeEach(() => (wrapper = setup()));

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render a title', () => {
    const title = findByDataTest(wrapper, 'title');
    expect(title.length).toBe(1);
  });

  it('Should render 3 reminders', () => {
    const reminders = findByDataTest(wrapper, 'reminder');
    expect(reminders.length).toBe(3);
  });
});

describe('ReminderGroup renders correctly without reminders', () => {
  it('Should render a no reminders message', () => {
    const props = {
      reminders: [],
      addReminder: testFunc,
      deleteReminder: testFunc
    };

    const wrapper = setup(props);
    const message = findByDataTest(wrapper, 'no-reminders-msg');
    expect(message.length).toBe(1);
  });
});

describe('ReminderGroup has correct props', () => {
  it('Should not return a prop error', () => {
    const propsError = checkProps(ReminderGroup, initialProps);
    expect(propsError).toBeUndefined();
  });
});
