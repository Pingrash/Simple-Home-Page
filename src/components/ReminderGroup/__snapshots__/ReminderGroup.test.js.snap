// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should match snapshot 1`] = `
<div>
  <h3
    className="MuiTypography-root MuiTypography-h4"
    data-test="title"
  >
    Your Reminders
  </h3>
  <div
    className="MuiPaper-root makeStyles-container-32 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h6
      className="MuiTypography-root makeStyles-title-33 MuiTypography-h6"
      data-test="title"
    >
      title1
    </h6>
    <p
      className="MuiTypography-root makeStyles-body-34 MuiTypography-body1"
      data-test="body"
    >
      body1
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-35 MuiButton-textSecondary"
      data-test="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Delete
      </span>
    </button>
  </div>
  <div
    className="MuiPaper-root makeStyles-container-32 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h6
      className="MuiTypography-root makeStyles-title-33 MuiTypography-h6"
      data-test="title"
    >
      title2
    </h6>
    <p
      className="MuiTypography-root makeStyles-body-34 MuiTypography-body1"
      data-test="body"
    >
      body2
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-35 MuiButton-textSecondary"
      data-test="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Delete
      </span>
    </button>
  </div>
  <div
    className="MuiPaper-root makeStyles-container-32 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h6
      className="MuiTypography-root makeStyles-title-33 MuiTypography-h6"
      data-test="title"
    >
      title3
    </h6>
    <p
      className="MuiTypography-root makeStyles-body-34 MuiTypography-body1"
      data-test="body"
    >
      body3
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-35 MuiButton-textSecondary"
      data-test="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Delete
      </span>
    </button>
  </div>
</div>
`;
